### ISSUE TITLE
after viewsets and routers are implemented, Post List functionality is available when logged out

### ISSUE THREAD

**Issue by mirrorstage** [wsvincent/restapiswithdjango#1]
After completing the book, I was exploring various states of the API while logged in and logged out. I found that in my completed version of the Blog API project, the Post List API view (`localhost:8000/api/v1`) is accessible when logged out. The endpoint is `200 OK` and you can create a new post for any existing user.

I found this strange, since that functionality is explicitly removed in Chapter 6:

> We no longer see our Post List page. Instead we are greeted with an unfriendly HTTP 403 Forbidden status code since we are not logged in. And there are no forms in the browsable API to edit the data since we don’t have permission.    (citation: page 106 in my PDF)

I get `403` after completing Chapter 7. It's also what I get in the sections of Chapter 8 that come before implementing viewsets and routers (starting on page 151 in my PDF). But by the end of Chapter 8, the response at `/api/v1` is `200`. 

To test whether this is something I had accidentally done, I downloaded a clean copy of the repo. I checked each chapter individually in its own virtual environment. I found the same behaviors as described above. I also took a git branch of a clean copy of the Chapter 7 source and worked through Chapter 8 up to page 151. Endpoint `api/v1` is still `403` at that point.

As this seems like something one wouldn't want available on a blog site, I thought I'd point it out. 😄 

I've reviewed the code that you add in Chapter 8 after page 151, and I'm afraid I don't understand the relationships among the permissions and the viewset/router implementations well enough to offer a suggestion for a fix. But I'd love to better understand what's going on here.

**Comment by wsvincent** [comment]
Hi @mirrorstage , Belated thanks for pointing this out. You are quite right. Somehow the permissions are not running through the Viewset and I don't have the short answer (so you're in good company). I'll sort it out though and report back! -William

**Comment by mirrorstage** [comment]
@wsvincent Thanks! Also pleased to see that (I think) I gave you some material for the latest Django Chat episode.

**Comment by wsvincent** [comment]
Ha, yes you did. Still trying to sort out why one permission is overriding another...

**Comment by evadeflow** [comment]
It's a bit worse than that. Anonymous users can not only view existing posts, they can also create _new_ posts, and (as described in issue #35) they can specify any existing user in the author field. (Just pointing this out as a courtesy—_really_ like the book, it's been super illuminating.)

**Comment by foxdon13** [comment]
Hi,
According to Django documentation, 


> Note: when you set new permission classes through class attribute or decorators you're telling the view to ignore the default list set over the settings.py file.

Hence, the default permission configured in the settings.py got ignored.

To solve this problem we must import the IsAuthenticated in the views.py and place it in the permission_classes as follows

```
class PostViewSet(viewsets.ModelViewSet):
    queryset = Post.objects.all()
    serializer_class = PostSerializer
    permission_classes = [IsAuthenticated, IsAuthorOrReadOnly]
```

I hope this clears out the confusion of permission overriding and ignoring.

**Comment by krotla** [comment]
In available version of the book for Django 5.1 this problem still exist in code example (Chapter 10: Viewsets and Routers).
